src/main/php/org/bovigo/vfs/content/StringBasedFileContent.php

```diff
@@ -63,7 +63,7 @@ public function size(): int
     protected function doRead(int $offset, int $count): string
     {
         $data = substr($this->content, $offset, $count);
-        return (false === $data) ? '' : $data;
+        return (!$data) ? '' : $data;
```

Comment:
you are changing the behavior here. `false === $data` will result in other code than `!$data` I don't know all cases but I don't like the implicit casting that is triggered here. 
Take a look at this example: https://3v4l.org/bMIMi

Reply:
The example looks correct to me. A non-empty string is truthy, so the inverse of it should be false. Switched it back anyway, but had to add an extra doc to make phpstan pass. phpstan thinks it'll only ever return a string, possibly based on the [docs](https://www.php.net/manual/en/function.substr.php#refsect1-function.substr-changelog).